tycho_simulation_py/python/tycho_simulation_py/evm/utils.py

```diff
@@ -194,6 +200,8 @@ def get_storage_slot_at_key(key: Address, mapping_slot: int) -> int:
     key_bytes = bytes.fromhex(key[2:]).rjust(32, b"\0")
     mapping_slot_bytes = int.to_bytes(mapping_slot, 32, "big")
     slot_bytes = eth_utils.keccak(key_bytes + mapping_slot_bytes)
+    if lang == ContractCompiler.Vyper:
+        slot_bytes = eth_utils.keccak(mapping_slot_bytes + key_bytes)
```

Comment:
Would it make sense to move this logic directly to `ContractCompiler`? Could make things a bit more obvious but it's not necessary imo.